licence-checker/licence-checker.py

```diff
@@ -243,12 +254,12 @@ def check_paths(config, git_paths):
             results.excluded(filepath, "Path matches exclude pattern")
             continue
 
-        check_file_for_licence(config.licence, results, filepath)
+        check_file_for_licence(config, results, filepath)
 
     return results
 
 
-def check_file_for_licence(licence, results, filepath):
+def check_file_for_licence(config, results, filepath):
```

Comment:
Maybe it would be better to pass `licence` and `match_regex` separately here, to be the same as `check_file_with_style`. I think we should be consistent either way, and probably prefer the more explicit version (which guarantees we won't start using some other field of `config`).

As a plus, the `check_file_with_style` call site will probably fit on one line again :-)